src/ORM.php

```diff
@@ -221,13 +211,28 @@ public function getSchema(): SchemaInterface
         return $this->schema;
     }
 
-    /**
-     * @inheritdoc
-     */
+    /** @deprecated since Cycle ORM v1.8, this method will be removed in future releases. Use method with instead. */
     public function withHeap(HeapInterface $heap): ORMInterface
     {
+        return $this->with(null, null, $heap);
+    }
+
+    public function with(
+        ?SchemaInterface $schema = null,
+        ?FactoryInterface $factory = null,
+        ?HeapInterface $heap = null
+    ): ORMInterface {
         $orm = clone $this;
-        $orm->heap = $heap;
+
+        if ($schema) {
```

Comment:
```suggestion
        if ($schema === null) {
```
Please use direct comparision for non-boolean values